Framework/PythonInterface/plugins/algorithms/CylinderPaalmanPingsCorrection2.py

```diff
@@ -405,10 +405,10 @@ def _set_material(self, ws_name, chemical_formula, density_type, density):
             mat = builder.setFormula(chemical_formula).setMassDensity(density).build()
```

Comment:
Ah, perhaps it is better to do `density = ws.experimentInfo().sample().getMaterial().numberDensity`? This should ensure the calculation is done once.